File: Programming with C#/3. C# Object-Oriented Programming/04. Object-Oriented Programming Fundamental Principles – Part I/02. Human/Student.cs
using System;
using System.Linq;

class Student : Human
{
    private uint grade = 0;

    public Student(string firstName, string lastName, uint grade) : base(firstName, lastName)
    {
        this.Grade = grade;
    }

    public uint Grade
    {
        get
        {
            return this.grade;
        }
        set
        {
            if (value <= 0 || value > 12)
                throw new ArgumentException("Grade is not in range!");

            this.grade = value;
        }
    }

    public override string ToString()
    {
        return string.Format("Student: {0} {1}, Grade: {2}", this.FirstName, this.LastName, this.Grade);
    }
}

File: Web Design & Development/5. JavaScript OOP/11. Advanced OOP (Autumn Academy)/01. School - Prototypal OOP/scripts/Classes/Student.js
define(function(require) {

    // Include utility functions and parent class
    var utility = require('../Helper/utility.js');
    var Human = require('./Human.js')

    var Student = Human.extend({
        init: function(firstName, lastName, age, grade) {
            Human.init.apply(this, arguments);
            
            this.grade = grade;

            return this;
        },
        introduce: function() {
            return Human.introduce.apply(this) + ", Grade: " + this.grade;
        }
    });

    return Student;
})